packages/hop-node/src/db/TransfersDb.ts

```diff
@@ -36,8 +42,61 @@ export type Transfer = {
 }
 
 class TransfersDb extends BaseDb {
-  async update (transferId: string, data: Partial<Transfer>) {
-    return super.update(transferId, data)
+  ready = false
+
+  constructor (prefix: string, _namespace?: string) {
+    super(prefix, _namespace)
+
+    // this only needs to be ran once on start up to backfill timestamped keys.
+    // this function can be removed once all bonders update.
+    // timestamped keys (in addition to transferId as keys) are needed to filter
+    // leveldb read streams.
+    this.trackTimestampedKeys()
+      .then(() => {
+        this.ready = true
+        this.logger.debug('transfersDb ready')
+      })
+      .catch(this.logger.error)
+  }
+
+  private async tilReady (): Promise<boolean> {
+    if (this.ready) {
+      return true
+    }
+
+    await wait(100)
+    return this.tilReady()
+  }
+
+  async trackTimestampedKeys () {
+    const transfers = await this.getTransfers()
+    for (const transfer of transfers) {
+      await this.trackTimestampedKey(transfer)
+    }
+  }
+
+  async trackTimestampedKey (transfer: Partial<Transfer>) {
+    const transferId = transfer.transferId
+    const key = this.getTimestampedKey(transfer)
+    if (!key || !transferId) {
+      return
+    }
+    const exists = await this.getById(key)
+    if (!exists) {
+      await super.update(key, { transferId })
+    }
+  }
+
+  getTimestampedKey (transfer: Partial<Transfer>) {
+    if (transfer?.transferSentTimestamp && transfer?.transferSentIndex !== undefined) {
```

Comment:
With our latest `pure` event update PR, I believe this data will not exist immediately. In that case, is the timestamp ever added to the entry?

Reply:
yes it's checked and added during and `.update()`

![DeepinScreenshot_select-area_20211004124930](https://user-images.githubusercontent.com/168240/135915084-12f54005-ae3c-4dc7-87e9-dabbcc5cef3d.png)